### ISSUE TITLE
Feature Request: Implement Turbo Button Simulation

### ISSUE THREAD

**Issue by tfzhu** [rileytestut/Delta#339]
# Description

The addition of a Turbo Button feature to simulate rapid button presses when a designated button is held down. This is distinct from the existing "Hold Buttons" feature as it aims to mimic the physical turbo buttons found on some classic gaming controllers such as the NES.

# Use Case
The primary use case for the turbo button simulation is to enhance the gaming experience in fast-paced, shooting games where quick, repetitive button presses are necessary for firing in-game weapons. By enabling turbo mode on a chosen button (e.g., the "A" button), the game will register multiple presses when the button is held down continuously, as opposed to the current feature which only registers a single, continuous press.

# Benefits
**Improved Gameplay**: Players can focus on strategy and movement without being hindered by the physical demand of rapid button presses.
**Accessibility**: This feature could also serve to make games more accessible to individuals who may have difficulty pressing buttons repeatedly due to physical constraints.
**Customizable Gaming Experience**: Players could tailor their controller setup to their personal preference, potentially assigning turbo to different buttons depending on the game they're playing.

# Technical Consideration
To implement this feature, the following technical aspects will need to be considered:

**Button Mapping**: A method for users to assign the turbo feature to specific buttons.
**Timing Configuration**: An adjustable rate for how quickly the button presses are simulated.
**Visual Indication**: Providing a visual cue to the user when a button is in turbo mode.

# Example
The following pseudocode outlines a simple method of how the turbo feature might be implemented:

```C++
turboButton = "A"        // Assign the A button to turbo mode
turboEnabled = false     // A flag indicating whether turbo mode is active

function update() {
    if (pressed(turboButton)) {
        turboEnabled = !turboEnabled   // Toggle turbo mode on or off
    }
    
    if (turboEnabled) {
        simulateRapidPress(turboButton)
    }
}

function simulateRapidPress(button) {
    // Logic to simulate rapid on/off states of the button
    // This should take into account the desired rate of presses
}

// Additional code to handle the timing and visual indication
```

# Conclusion
In summary, adding a turbo button feature would enable gamers to experience enhanced gameplay in certain genres that demand rapid input, boost accessibility for a wide range of players, and allow for a more personalized gaming experience.
